controllers/status_controller.go

```diff
@@ -97,13 +97,14 @@ func (r *StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	log.V(debugLogLevel).Info("updating socket")
-	if owner.Spec.DedicatedLoadBalancerIP == nil {
+	if !owner.EnableDedicatedSVCLB() {
```

Comment:
In the old code you executed the `else` block when the value of the IP was an empty string (aka `""`) now you enter the `if` block . this one looks more correct. i hope  this is compatible with existing specs.